invokeai/app/services/model_records/model_records_sql.py

```diff
@@ -96,10 +96,11 @@ def _create_tables(self) -> None:
             CREATE TABLE IF NOT EXISTS model_config (
                 id TEXT NOT NULL PRIMARY KEY,
                 -- The next 3 fields are enums in python, unrestricted string here
-                base TEXT NOT NULL,
-                type TEXT NOT NULL,
-                name TEXT NOT NULL,
-                path TEXT NOT NULL,
+                base TEXT GENERATED ALWAYS as (json_extract(config, '$.base')) VIRTUAL NOT NULL,
+                type TEXT GENERATED ALWAYS as (json_extract(config, '$.type')) VIRTUAL NOT NULL,
+                name TEXT GENERATED ALWAYS as (json_extract(config, '$.name')) VIRTUAL NOT NULL,
+                path TEXT GENERATED ALWAYS as (json_extract(config, '$.path')) VIRTUAL NOT NULL,
+                format TEXT GENERATED ALWAYS as (json_extract(config, '$.format')) VIRTUAL NOT NULL,
```

Comment:
When we get the migrator merged, we'll need to remember to do a `DROP TABLE IF EXISTS` before creating this table, as it has changed since 3.4.0 was released.

Reply:
Is it safe to add a migrator to the migrator PR at this point? I don’t think any of the proposed changes to the version comparisons or backup strategy would affect that.

Reply:
I've just added it to the migrator PR: https://github.com/invoke-ai/InvokeAI/pull/5227/files#diff-85498dbec06fc5f334e21a3a7393eac1cfb65c58dfabf884fb499f04f8a0f25aR105-R137